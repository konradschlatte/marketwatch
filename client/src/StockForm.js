import * as React from 'react';
import Chart from './Chart';
const axios = require('axios');

export default class StockForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      stock: '',
      period: '1d',
      list: [],
      id: 0,
      counter: 0
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    this.getDataFromDb();
  };

  getDataFromDb = () => {
    fetch('http://localhost:3001/getData')
      .then((data) => data.json())
      .then((res) => this.setState({ list: res.data }));
  };

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;
    this.setState((state) => {
      return { [name]: value, id: this.state.id + 1 };
    });
  };

  onAddItem = () => {
    let id = this.state.id;
    let stock = this.state.stock;
    let period = this.state.period;

    axios.post('http://localhost:3001/putData', {
      id: id,
      stock: stock,
      period: period
    });
    const values = this.state.list.concat({ stock: stock, period: period, id: id });
    this.setState((state) => {
    return { list: values };
    });
  };

  handleSubmit(event) {
    event.preventDefault();
    this.onAddItem();
  }

  deleteChart = (key) => {
    let charts = this.state.list.filter(item => item.id !== key);
    this.setState({ list: charts });
  };

  render() {
    return (
      <section>
      <form onSubmit={this.handleSubmit}>
        <label>
        Stock:
        </label>
          <input
            id="first"
            name="stock"
            type="text"
            placeholder = "Symbol"
            value={this.state.value}
            onChange={this.handleInputChange} />

        <label>
          Period:
        </label>
            <select name = "period" type = "select" value={this.state.value} onChange={this.handleInputChange}>
              <option value="1d">1 day</option>
              <option value="1m">1 month</option>
              <option value="3m">3 months</option>
              <option value="6m">6 months</option>
              </select>

        <input id="submit" type="submit" value="submit"/>
      </form>
      <div>
      { (this.state.list || []).map(item => (
          <Chart key = { item.id } delEvent = {() => this.deleteChart(item.id) }
            uri = { 'https://api.iextrading.com/1.0/stock/' + item.stock.toUpperCase() + '/chart/' + item.period }
            legend = { item.stock + ' ' + item.period }/>
        )) }
       </div>
      </section>
    );
  }
}
