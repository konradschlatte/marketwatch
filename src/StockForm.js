import * as React from 'react';
import Chart from './Graph';
//import './index.css';

export default class StockForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      stock: '',
      period: '1d',
      list: []
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;
    this.setState((state) => {
    // Important: read `state` instead of `this.state` when updating.
    return { [name]: value };
    // this.setState({
    //   [name]: value
    // });
    });
  };
  onAddItem = () => {
    const values = this.state.list.concat({ stock: this.state.stock, period: this.state.period});
    this.setState((state) => {
    return { list: values};
    });
  };

  handleSubmit(event) {
    event.preventDefault();
    this.onAddItem();
  //  chartCall(this.state.list);
  }

  render() {
    return (
      <section>
      <form onSubmit={this.handleSubmit}>
        <label>
        Stock:
        </label>
          <input
            id="first"
            name="stock"
            type="text"
            placeholder = "Symbol"
            value={this.state.value}
            onChange={this.handleInputChange} />

        <label>
          Period:
        </label>
            <select name = "period" type = "select" value={this.state.value} onChange={this.handleInputChange}>
              <option value="1d">1 day</option>
              <option value="1m">1 month</option>
              <option value="3m">3 months</option>
              <option value="6m">6 months</option>
              </select>

        <input id="submit" type="submit" value="submit"/>
      </form>
      <div>
      { (this.state.list || []).map((item, i) => (
          <Chart key={i} uri = { 'https://api.iextrading.com/1.0/stock/' + item.stock.toUpperCase() + '/chart/' + item.period } legend = { item.stock + ' ' + item.period }/>
        )) }
       </div>
      </section>
    );
  }
}
