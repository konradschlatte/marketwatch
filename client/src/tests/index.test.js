import React from 'react';
import { render } from 'react-dom';
import ReactDOM from 'react-dom';
import StockForm from '../StockForm';
import renderer from 'react-test-renderer';
import loadForm from '../index';
jest.mock('react-chartjs-2', () => ({ Line: () => <div>Chart</div> }));

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<StockForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});
test('StockForm has not changed appearance', () => {
  const component = renderer.create(
    <StockForm />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
jest.mock('react-dom');
describe('loadForm ', () => {
  test('it calls ReactDom.render', () => {
    loadForm();
    expect(render).toHaveBeenCalledTimes(2);
  });
});
