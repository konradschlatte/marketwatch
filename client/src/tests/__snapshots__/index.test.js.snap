// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockForm has not changed appearance 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <label>
      Stock:
    </label>
    <input
      id="first"
      name="stock"
      onChange={[Function]}
      placeholder="Symbol"
      type="text"
    />
    <label>
      Period:
    </label>
    <select
      id="period"
      name="period"
      onChange={[Function]}
      type="select"
    >
      <option
        value="1d"
      >
        1 day
      </option>
      <option
        value="1m"
      >
        1 month
      </option>
      <option
        value="3m"
      >
        3 months
      </option>
      <option
        value="6m"
      >
        6 months
      </option>
    </select>
    <label>
      MA1:
    </label>
    <select
      id="ma1"
      name="ma1"
      onChange={[Function]}
      type="select"
    >
      <option
        value="10"
      >
        10 day
      </option>
      <option
        value="30"
      >
        30 day
      </option>
      <option
        value="60"
      >
        60 day
      </option>
    </select>
    <label>
      MA2:
    </label>
    <select
      id="ma2"
      name="ma2"
      onChange={[Function]}
      type="select"
    >
      <option
        value="10"
      >
        10 day
      </option>
      <option
        value="30"
      >
        30 day
      </option>
      <option
        value="60"
      >
        60 day
      </option>
    </select>
    <input
      id="submit"
      type="submit"
      value="submit"
    />
  </form>
  <div />
</section>
`;
